sdk/metric/meter_example_test.go

```diff
@@ -0,0 +1,371 @@
+					IsMonotonic: true,
+					DataPoints: []metricdata.DataPoint[int64]{
+						{
+							Attributes: filteredAttributeSet,
+							Value:      80,
+						},
+					},
+				},
+			},
+			{
+				Name: "pageFaults",
+				Data: metricdata.Sum[int64]{
+					Temporality: metricdata.CumulativeTemporality,
+					IsMonotonic: true,
+					DataPoints: []metricdata.DataPoint[int64]{
+						{
+							Attributes: attribute.NewSet(tid1Attrs...),
+							Value:      50,
+						},
+						{
+							Attributes: attribute.NewSet(tid2Attrs...),
+							Value:      30,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	metrics, err := reader.Collect(ctx)
+	assert.NoError(t, err)
+	metricdatatest.AssertEqual(t, wantScopeMetrics, metrics.ScopeMetrics[0], metricdatatest.IgnoreTimestamp())
+
+	wantFilterValue := &wantScopeMetrics.Metrics[0].Data.(metricdata.Sum[int64]).DataPoints[0].Value
+	wantDataPoint1Value := &wantScopeMetrics.Metrics[1].Data.(metricdata.Sum[int64]).DataPoints[0].Value
+	wantDataPoint2Value := &wantScopeMetrics.Metrics[1].Data.(metricdata.Sum[int64]).DataPoints[1].Value
+
+	// During the time range (T1, T2]:
+	//     pid = 1001, tid = 1, #PF = 53
+	//     pid = 1001, tid = 2, #PF = 38
+
+	atomic.StoreInt64(&pfValues[0], 53)
+	atomic.StoreInt64(&pfValues[1], 38)
+
+	*wantFilterValue = 91
+	*wantDataPoint1Value = 53
+	*wantDataPoint2Value = 38
+
+	metrics, err = reader.Collect(ctx)
+	assert.NoError(t, err)
+	metricdatatest.AssertEqual(t, wantScopeMetrics, metrics.ScopeMetrics[0], metricdatatest.IgnoreTimestamp())
+
+	// During the time range (T2, T3]
+	//     pid = 1001, tid = 1, #PF = 56
+	//     pid = 1001, tid = 2, #PF = 42
+
+	atomic.StoreInt64(&pfValues[0], 56)
+	atomic.StoreInt64(&pfValues[1], 42)
+
+	*wantFilterValue = 98
+	*wantDataPoint1Value = 56
+	*wantDataPoint2Value = 42
+
+	metrics, err = reader.Collect(ctx)
+	assert.NoError(t, err)
+	metricdatatest.AssertEqual(t, wantScopeMetrics, metrics.ScopeMetrics[0], metricdatatest.IgnoreTimestamp())
+
+	// During the time range (T3, T4]:
+	//     pid = 1001, tid = 1, #PF = 60
+	//     pid = 1001, tid = 2, #PF = 47
+
+	atomic.StoreInt64(&pfValues[0], 60)
+	atomic.StoreInt64(&pfValues[1], 47)
+
+	*wantFilterValue = 107
+	*wantDataPoint1Value = 60
+	*wantDataPoint2Value = 47
+
+	metrics, err = reader.Collect(ctx)
+	assert.NoError(t, err)
+	metricdatatest.AssertEqual(t, wantScopeMetrics, metrics.ScopeMetrics[0], metricdatatest.IgnoreTimestamp())
+
+	// During the time range (T4, T5]:
+	//     thread 1 died, thread 3 started
+	//     pid = 1001, tid = 2, #PF = 53
+	//     pid = 1001, tid = 3, #PF = 5
+
+	atomic.StoreInt64(&pfValues[0], 0)
+	atomic.StoreInt64(&pfValues[1], 53)
+	atomic.StoreInt64(&pfValues[2], 5)
+
+	*wantFilterValue = 58
+	wantAgg := metricdata.Sum[int64]{
+		Temporality: metricdata.CumulativeTemporality,
+		IsMonotonic: true,
+		DataPoints: []metricdata.DataPoint[int64]{
+			{
+				Attributes: attribute.NewSet(tid1Attrs...),
+				Value:      60,
```

Comment:
Should this be 0?

Reply:
I think it would be if we dropped cumulative values, which we don't. When `pfValues[0]` is `0` it means it is not observed: https://github.com/MrAlias/opentelemetry-go/pull/721/files/79e0f4adce2c68f6c3efeadc50a65538ae325e55#diff-7ddf00e1a0bfde5e78e6aaa862ad8120cdacfde1c0eeb04ddc106242c294cde3R62

There are two issues in the specification that are tracking this work:

- https://github.com/open-telemetry/opentelemetry-specification/issues/3037
- https://github.com/open-telemetry/opentelemetry-specification/issues/1891